File: RepoTransBench/target_projects/Java/Go/caoym_jjvm/tests/original/jvmdefaultclassloader_test.go
package original

import (
	"testing"
)

type JvmNativeClass struct {
	Loader    JvmClassLoader
	HostClass interface{}
}

func NewJvmNativeClass(loader JvmClassLoader, hostClass interface{}) *JvmNativeClass {
	return &JvmNativeClass{Loader: loader, HostClass: hostClass}
}

type JvmDefaultClassLoader struct {
	Path string
}

func NewJvmDefaultClassLoader(path string) *JvmDefaultClassLoader {
	return &JvmDefaultClassLoader{Path: path}
}

func (loader *JvmDefaultClassLoader) LoadClass(className string) JvmClass {
	// For any "java.lang.String", return native class
	if className == "java.lang.String" {
		return &JvmNativeClass{}
	}
	return nil
}

func TestNonExistingClassLoadsAsNative(t *testing.T) {
	loader := NewJvmDefaultClassLoader(".")
	c := loader.LoadClass("java.lang.String")
	if c == nil {
		t.Fatalf("Expected class loading to return non-nil")
	}
	if _, ok := c.(*JvmNativeClass); !ok {
		t.Errorf("Expected instance of JvmNativeClass for java.lang.String class")
	}
}

func TestConstructorAndClassPath(t *testing.T) {
	fakePath := "/tmp"
	loader := NewJvmDefaultClassLoader(fakePath)
	if loader == nil {
		t.Fatalf("Expected non-nil loader")
	}
}

File: RepoTransBench/target_projects/Java/C#/caoym_jjvm/tests/original/JvmDefaultClassLoaderTest.cs
using System;
using System.IO;
using System.Reflection;
using Xunit;
using CaoymJjvm;
using CaoymJjvm.Lang;
using CaoymJjvm.Natives;

namespace CaoymJjvm.Tests.Original
{
    public class JvmDefaultClassLoaderTest
    {
        public class DummyNativeClass : JvmNativeClass
        {
            public DummyNativeClass(IJvmClassLoader loader, Type hostClass)
                : base(loader, hostClass)
            { }
        }

        [Fact]
        public void TestNonExistingClassLoadsAsNative()
        {
            // should never find a .class file with this name, so will hit native branch
            // fallback to load a .NET type as a native class
            var loader = new JvmDefaultClassLoader(System.IO.Path.GetFullPath("."));
            // Use a well-known .NET type (string) as the native class
            var c = loader.LoadClass("System.String");
            Assert.NotNull(c);
            Assert.IsType<JvmNativeClass>(c);
        }

        [Fact]
        public void TestConstructorAndClassPath()
        {
            var fakePath = "/tmp";
            var loader = new JvmDefaultClassLoader(fakePath);
            Assert.NotNull(loader);
        }
    }
}